components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -2005,6 +2008,137 @@ void JsonRpcService::GetERC1155TokenBalance(
+          mojom::ProviderError::kInvalidParams,
+          l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+      return;
+    }
+  }
+
+  auto internal_callback = base::BindOnce(
+      &JsonRpcService::OnGetAllTokensDiscoverAssets,
+      weak_ptr_factory_.GetWeakPtr(), chain_id, account_addresses,
+      std::move(user_assets), std::move(callback));
+
+  BlockchainRegistry::GetInstance()->GetAllTokens(
+      chain_id, mojom::CoinType::ETH, std::move(internal_callback));
+  return;
+}
+
+void JsonRpcService::OnGetAllTokensDiscoverAssets(
+    const std::string& chain_id,
+    const std::vector<std::string>& account_addresses,
+    std::vector<mojom::BlockchainTokenPtr> user_assets,
+    DiscoverAssetsCallback callback,
+    std::vector<mojom::BlockchainTokenPtr> token_registry) {
+  auto network_url = GetNetworkURL(prefs_, chain_id, mojom::CoinType::ETH);
+  if (!network_url.is_valid()) {
+    std::move(callback).Run(
+        std::vector<mojom::BlockchainTokenPtr>(),
+        mojom::ProviderError::kInvalidParams,
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+
+  base::Value topics(base::Value::Type::LIST);
+  if (!MakeAssetDiscoveryTopics(account_addresses, &topics)) {
+    std::move(callback).Run(
+        std::vector<mojom::BlockchainTokenPtr>(),
+        mojom::ProviderError::kInvalidParams,
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+
+  // Create list of contract addresses to search by
+  // removing non erc20 tokens and assets that user has already
+  std::unordered_set<std::string> user_asset_contract_addresses;
+  for (auto& user_asset : user_assets) {
+    user_asset_contract_addresses.insert(user_asset->contract_address);
+  }
+  base::Value contract_addresses(base::Value::Type::LIST);
+  for (const auto& registry_token : token_registry) {
+    if (registry_token->is_erc20 && !registry_token->contract_address.empty() &&
+        user_asset_contract_addresses.find(registry_token->contract_address) ==
+            user_asset_contract_addresses.end()) {
+      contract_addresses.Append(registry_token->contract_address);
+    }
+  }
+
+  if (contract_addresses.GetList().size() == 0) {
+    std::move(callback).Run(std::vector<mojom::BlockchainTokenPtr>(),
+                            mojom::ProviderError::kSuccess, "");
+    return;
+  }
+
+  auto internal_callback =
+      base::BindOnce(&JsonRpcService::OnGetTransferLogs,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  RequestInternal(
+      eth::eth_getLogs("earliest", "latest", &contract_addresses, &topics, ""),
+      true, network_url, std::move(internal_callback));
+  return;
+}
+
+void JsonRpcService::OnGetTransferLogs(
+    DiscoverAssetsCallback callback,
+    const int status,
+    const std::string& body,
+    const base::flat_map<std::string, std::string>& headers) {
+  std::vector<Log> logs;
+  if (!eth::ParseEthGetLogs(body, &logs)) {
```

Comment:
I believe we need to sanitize the JSON input here before we parse it since it's still in an untrusted input from what I can tell

Reply:
It's already sanitized -- RequestInternal uses the api_request_helper which runs all response through the sanitizer

Reply:
My mistake, you're correct. Thanks for pointing that portion of the code out.